CPack RPM packaging (#3601)
ElektraInitiative/libelektra: doc/INSTALL.md

First follow the steps in [COMPILE](COMPILE.md).

After you completed building Elektra on your own, there are multiple options how to install it. For example, with make or CPack tools.
We recommend that you generate your own packages with CPack so ensure compatibility with future releases.

### CPack

The current supported systems are: Debian, Ubuntu and Fedora.

First follow the steps in [COMPILE](COMPILE.md).

Then use:

```sh
make package
```

which will create a package for distributions where a Generator is implemented.

> You can find the generated packages in the `packages` directory of the build directory.

Reviewer: Maybe we can rename this folder to "package" as, at least zsh, always suggests to correct package to packages.

First follow the steps in [COMPILE](COMPILE.md).

After you completed building Elektra on your own, there are multiple options how to install it. For example, with make or CPack tools.

> We recommend that you generate your own packages with CPack so ensure compatibility with future releases.

Reviewer: ```suggestion
We recommend to use the packages from our build server or that you generate your own packages with CPack.
```

### CPack

The current supported systems are: Debian, Ubuntu and Fedora.

Then use:

```sh
make package
```

which will create packages for distributions where a Generator is implemented.

You can find the generated packages in the `package` directory of the build directory.

> NOTE: If all plugins/bindings/tools a package includes are excluded, the package will be not generated.

#### Debian/Ubuntu

On Debian based distributions you will need to set LD_LIBRARY_PATH before generating the package.
Simply `cd` into the build directory and run following command:

```sh
LD_LIBRARY_PATH=$(pwd)/lib:${LD_LIBRARY_PATH} make package
```

To install the packages run this in the `package` directory:

```sh

> dpkg -i *

Reviewer: With `apt` you can install directly with also installing dependencies.

xargs rm < install_manifest.txt
```

### CPack

> First follow the steps in [COMPILE](COMPILE.md).

Reviewer: Maybe we should keep this info somewhere?

Author: It is already present in the parent heading `## OS Independent` (not visible in the git diff). Is it really necessary to write this multiple times?

Reviewer: Yes, you are right, I did not see it.
